src/aero_state.F90

```diff
@@ -2867,28 +3193,37 @@ subroutine aero_state_input_netcdf(aero_state, ncid, aero_data)
 
     do i_part = 1,n_part
        aero_particle%vol = aero_particle_mass(i_part, :) / aero_data%density
-       aero_particle%n_orig_part = aero_n_orig_part(i_part, :)
+       allocate(aero_particle%component(aero_component_len(i_part)))
+       do i_comp = 1,aero_component_len(i_part)
+          aero_particle%component(i_comp)%source_id = &
+               aero_component_source_num(aero_component_start_ind(i_part) &
+               + i_comp - 1)
+          aero_particle%component(i_comp)%create_time = &
+               aero_component_create_time(aero_component_start_ind(i_part) &
+               + i_comp - 1)
+       end do
        aero_particle%weight_group = aero_particle_weight_group(i_part)
        aero_particle%weight_class = aero_particle_weight_class(i_part)
-       if (size(aero_absorb_cross_sect) == n_part) then
-          aero_particle%absorb_cross_sect = aero_absorb_cross_sect(i_part)
+       if (size(aero_absorb_cross_sect, 1) == n_part) then
```

Comment:
When you do not compute optical properties in a simulation, the netCDF output file does not contain these variables (as they'd all be zeros). When you read in a netCDF file, it will attempt to find these optical values to repopulate particle properties but it will not find them. It handlers this special situation by allocating these arrays as empty.

This `if` statement here checks to make sure the array is the correct size, which means that it should assign it to the particles.